### ISSUE TITLE
Could you please add a node for the "Face Color Enhancement and Restoration" feature, which allows for the repair and colorization of old photos, to the CodeFormer project? 

### ISSUE THREAD

**Issue by ericlee2023** [mav-rik/facerestore_cf#23]
Could you please add a node for the "Face Color Enhancement and Restoration" feature, which allows for the repair and colorization of old photos, to the CodeFormer project? Currently, I only see a "Face Restoration" node available.
![Snipaste_2024-05-29_23-53-01](https://github.com/mav-rik/facerestore_cf/assets/133652787/1801f032-beb9-4d13-af07-949f52dcfeef)

**Comment by zslefour** [comment]
You can add the following code to your _init_.py to do what you want.
=============================================

class FaceColorModelLoader:
    @classmethod
    def INPUT_TYPES(s):
        return {"required": { "model_name": (folder_paths.get_filename_list("facerestore_models"), ),
                             }}
    RETURN_TYPES = ("FACECOLOR_MODEL",)
    FUNCTION = "load_model"

    CATEGORY = "facerestore_cf"

    # def load_model(self, model_name):
    #     model_path = folder_paths.get_full_path("facerestore_models", model_name)
    #     sd = comfy.utils.load_torch_file(model_path, safe_load=True)
    #     out = model_loading.load_state_dict(sd).eval()
    #     return (out, )

    def load_model(self, model_name):
        if "codeformer" in model_name.lower():
            print(f'\tLoading CodeFormer: {model_name}')
            model_path = folder_paths.get_full_path("facerestore_models", model_name)
            device = model_management.get_torch_device()
            codeformer_net = ARCH_REGISTRY.get("CodeFormer")(
                dim_embd=512,
                codebook_size=1024,
                n_head=8,
                n_layers=9,
                connect_list=["32", "64", "128"],
            ).to(device)
            checkpoint = torch.load(model_path)["params_ema"]
            codeformer_net.load_state_dict(checkpoint)
            out = codeformer_net.eval()  
            return (out, )
        else:
            model_path = folder_paths.get_full_path("facerestore_models", model_name)
            sd = comfy.utils.load_torch_file(model_path, safe_load=True)
            out = model_loading.load_state_dict(sd).eval()
            return (out, )

class FaceColorWithModel:
    @classmethod
    def INPUT_TYPES(s):
        return {"required": { "facecolor_model": ("FACECOLOR_MODEL",),
                              "image": ("IMAGE",),
                              "facedetection": (["retinaface_resnet50", "retinaface_mobile0.25", "YOLOv5l", "YOLOv5n"],),
                              "codeformer_fidelity": ("FLOAT", {"default": 0.5, "min": 0.0, "max": 1, "step": 0.05})
                              }}

    RETURN_TYPES = ("IMAGE",)

    FUNCTION = "color_face"

    CATEGORY = "facerestore_cf"

    def __init__(self):
        self.face_helper = None

    def color_face(self, facecolor_model, image, facedetection, codeformer_fidelity):
        print(f'\tStarting restore_face with codeformer_fidelity: {codeformer_fidelity}')
        device = model_management.get_torch_device()
        facecolor_model.to(device)
        if self.face_helper is None:
            self.face_helper = FaceRestoreHelper(1, face_size=512, crop_ratio=(1, 1), det_model=facedetection, save_ext='png', use_parse=True, device=device)

        image_np = 255. * image.cpu().numpy()

        total_images = image_np.shape[0]
        out_images = np.ndarray(shape=image_np.shape)

        for i in range(total_images):
            cur_image_np = image_np[i,:, :, ::-1]

            original_resolution = cur_image_np.shape[0:2]

            if facecolor_model is None or self.face_helper is None:
                return image

            self.face_helper.clean_all()
            self.face_helper.read_image(cur_image_np)
            self.face_helper.get_face_landmarks_5(only_center_face=False, resize=640, eye_dist_threshold=5)
            self.face_helper.align_warp_face()

            restored_face = None
            for idx, cropped_face in enumerate(self.face_helper.cropped_faces):
                cropped_face_t = img2tensor(cropped_face / 255., bgr2rgb=False, float32=True)
                normalize(cropped_face_t, (0.5, 0.5, 0.5), (0.5, 0.5, 0.5), inplace=True)
                cropped_face_t = cropped_face_t.unsqueeze(0).to(device)

                try:
                    with torch.no_grad():
                        #output = facecolor_model(cropped_face_t, w=strength, adain=True)[0]
                        # output = facecolor_model(cropped_face_t)[0]
                        output = facecolor_model(cropped_face_t)[0]
                        restored_face = tensor2img(output, rgb2bgr=True, min_max=(-1, 1))
                    del output
                    torch.cuda.empty_cache()
                except Exception as error:
                    print(f'\tFailed inference for CodeFormer: {error}', file=sys.stderr)
                    restored_face = tensor2img(cropped_face_t, rgb2bgr=True, min_max=(-1, 1))

                restored_face = restored_face.astype('uint8')
                self.face_helper.add_restored_face(restored_face)

            self.face_helper.get_inverse_affine(None)

            restored_img = self.face_helper.paste_faces_to_input_image()
            restored_img = restored_img[:, :, ::-1]

            if original_resolution != restored_img.shape[0:2]:
                restored_img = cv2.resize(restored_img, (0, 0), fx=original_resolution[1]/restored_img.shape[1], fy=original_resolution[0]/restored_img.shape[0], interpolation=cv2.INTER_LINEAR)

            self.face_helper.clean_all()

            # restored_img = cv2.cvtColor(restored_face, cv2.COLOR_BGR2RGB)

            out_images[i] = restored_img

        restored_img_np = np.array(out_images).astype(np.float32) / 255.0
        restored_img_tensor = torch.from_numpy(restored_img_np)
        return (restored_img_tensor,)


NODE_CLASS_MAPPINGS = {
    "FaceRestoreCFWithModel": FaceRestoreCFWithModel,
    "CropFace": CropFace,
    "FaceRestoreModelLoader": FaceRestoreModelLoader,
    "FaceColorModelLoader": FaceColorModelLoader,
    "FaceColorWithModel": FaceColorWithModel,
}

**Comment by zslefour** [comment]
![image](https://github.com/user-attachments/assets/c228f3ec-4540-4fad-b2d5-77fabb1afe94)
